prboom2/src/SDL/i_video.c

```diff
@@ -810,28 +825,31 @@ static void I_FillScreenResolutionsList(void)
     screen_resolutions_list[list_size] = NULL;
   }
 
-  if (list_size == 0)
-  {
-    doom_snprintf(mode_name, sizeof(mode_name), "%dx%d", desired_screenwidth, desired_screenheight);
-    screen_resolutions_list[0] = Z_Strdup(mode_name);
-    current_resolution_index = 0;
-    list_size = 1;
-  }
+  // [FG] if the desired resolution not in the list, append it
```

Comment:
This comment should rather be on top of `if (current_resolution_index == -1)`, right?